From 341fe8143926c2dda50bb0ae931ef2e14bcbe47d Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Mon, 12 Jun 2023 21:42:44 -0700
Subject: [PATCH] remove xen build for STX

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 debian/control | 14 --------------
 debian/rules   | 33 ---------------------------------
 2 files changed, 47 deletions(-)

diff --git a/debian/control b/debian/control
index adf8946..d0fb7e8 100644
--- a/debian/control
+++ b/debian/control
@@ -247,20 +247,6 @@ Description: QEMU full system emulation binaries (x86)
  On x86 host hardware this package also enables KVM kernel virtual machine
  usage on systems which supports it.
 
-Package: qemu-system-xen
-Architecture: amd64
-Multi-Arch: no
-# do we really need qemu-system-data? keymaps only?
-Depends: ${shlibs:Depends}, ${misc:Depends}, qemu-system-data (>> ${source:Upstream-Version}~),
- seabios, ipxe-qemu
-Recommends: qemu-utils,
- ovmf,
-Description: QEMU full system emulation (Xen helper package)
- This package provides the i386 system emulation binary to work
- together with the Xen hypervisor for some types of DomUs.
- This package is not useful by its own.
-
-
 Package: qemu-utils
 Architecture: amd64 arm arm64 armel armhf hppa i386 ia64 kfreebsd-amd64 kfreebsd-i386 mips mipsel mips64 mips64el powerpc powerpcspe ppc64 ppc64el riscv64 s390x sparc sparc64 x32
 Multi-Arch: foreign
diff --git a/debian/rules b/debian/rules
index 78c1296..b0fd791 100755
--- a/debian/rules
+++ b/debian/rules
@@ -40,7 +40,6 @@ BUILD_PACKAGES := $(shell dh_listpackages)
 enable_system = $(if $(filter qemu-system,${BUILD_PACKAGES}),enable,disable)
 enable_linux_user = $(if $(filter qemu-user,${BUILD_PACKAGES}),enable,disable)
 
-QEMU_XEN = /usr/libexec/xen-qemu-system-i386
 PKGVERSION = Debian ${DEB_VERSION}
 SAVEMODDIR = /run/qemu/$(shell echo -n "${PKGVERSION}" | tr --complement '[:alnum:]+-.~' '_')
 sysdataidir = debian/qemu-system-data/usr/share/qemu
@@ -306,38 +305,6 @@ ifeq ($(filter-out $(DEB_HOST_ARCH),amd64)${enable_system},enable)
 qemu-builds += microvm
 endif
 
-##############################################
-# xen build (amd64 arch only, i386-softmmu target only)
-configure-xen: b/xen/configured
-b/xen/configured: configure
-	# system build for qemu-system-xen
-	rm -rf b/xen; mkdir -p b/xen
-	cd b/xen && \
-	    ../../configure ${common_configure_opts} \
-		--disable-blobs --disable-docs --disable-tools \
-		--without-default-features \
-		--enable-avx2 --enable-avx512f \
-		--enable-xen --target-list=i386-softmmu \
-		--enable-xen-pci-passthrough \
-		--disable-tcg --disable-kvm \
-		--audio-drv-list= \
-		--enable-libusb \
-		--enable-vnc --enable-vnc-jpeg \
-		--enable-spice \
-		--enable-virtfs --enable-attr --enable-cap-ng \
-		${QEMU_XEN_CONFIGURE_OPTIONS}
-	touch $@
-build-xen: b/xen/built
-b/xen/built: b/xen/configured
-	$(MAKE) -C b/xen V=${V} qemu-system-i386
-	touch $@
-install-xen: b/xen/built
-	install -D b/xen/qemu-system-i386 \
-	  debian/qemu-system-xen${QEMU_XEN}
-ifeq (${DEB_HOST_ARCH}-${enable_system},amd64-enable)
-qemu-builds += xen
-endif
-
 ##############################################
 # linux-user-static build
 configure-user-static: b/user-static/configured
-- 
2.25.1

